crates/types/src/data.rs

```diff
@@ -627,40 +634,22 @@ pub struct SequencingLeaf<TYPES: NodeType> {
 
 impl<TYPES: NodeType> PartialEq for SequencingLeaf<TYPES> {
     fn eq(&self, other: &Self) -> bool {
-        let delta_left = match &self.deltas {
-            Either::Left(deltas) => (deltas.0, deltas.1.commit()),
-            Either::Right(deltas) => (deltas.block_number(), deltas.payload_commitment()),
-        };
-        let delta_right = match &other.deltas {
-            Either::Left(deltas) => (deltas.0, deltas.1.commit()),
-            Either::Right(deltas) => (deltas.block_number(), deltas.payload_commitment()),
-        };
         self.view_number == other.view_number
-            // && self.height == other.height
             && self.justify_qc == other.justify_qc
             && self.parent_commitment == other.parent_commitment
-            && delta_left == delta_right
+            && self.block_header == other.block_header
+            && self.transaction_commitments == other.transaction_commitments
```

Comment:
Any reason this can't be derived?

Reply:
I think it couldn't be simply derived due to the `deltas` field, but it's not the case anymore. I'll update it.

Reply:
Ah, now that we have an optional payload, it should be skipped in the comparison.

Reply:
Oh right. That's a bit annoying but oh well